File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperClienteGPS.java
package com.friendlypos.principal.helpers;

import android.app.Activity;
import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver;
import com.friendlypos.application.datamanager.BaseManager;
import com.friendlypos.application.interfaces.RequestInterface;
import com.friendlypos.login.util.SessionPrefes;
import com.friendlypos.preventas.modelo.EnviarClienteVisitado;
import com.friendlypos.preventas.modelo.visit;
import com.friendlypos.principal.activity.MenuPrincipal;
import com.friendlypos.principal.modelo.EnviarClienteGPS;
import com.friendlypos.principal.modelo.customer_location;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static android.content.ContentValues.TAG;

/**
 * Created by DelvoM on 24/10/2018.
 */

public class SubirHelperClienteGPS {

    private NetworkStateChangeReceiver networkStateChangeReceiver;
    private MenuPrincipal activity;
    private Context mContext;
    private RequestInterface mAPIService;

    int codigo;
    public String respuestaServer;
    String codigoS;
    String mensajeS;
    String resultS;

    public SubirHelperClienteGPS(MenuPrincipal activity) {
        this.activity = activity;
        this.mContext = activity;
        networkStateChangeReceiver = new NetworkStateChangeReceiver();
        mAPIService = BaseManager.getApi();
    }

    public void sendPostClienteGPS(EnviarClienteGPS facturaQuery) {
        String token = "Bearer " + SessionPrefes.get(mContext).getToken();
        Log.d("tokenCliente", token + " ");
        if (isOnline()) {
            Log.d("factura1", facturaQuery + " ");
            mAPIService.savePostClienteGPS(facturaQuery, token).enqueue(new Callback<customer_location>() {

                @Override
                public void onResponse(Call<customer_location> call, Response<customer_location> response) {

                        if(response.isSuccessful()) {
                            // showResponse(response.body().toString());
                            Log.d("respVentaDirecta",response.body().toString());
                            codigo = response.code();
                            codigoS = response.body().getCode();
                            mensajeS = response.body().getMessage();
                            resultS= String.valueOf(response.body().isResult());

                            activity.codigoDeRespuestaClienteGPS(codigoS, mensajeS, resultS, codigo);
                        }
                        else{
                        }
                }

                @Override
                public void onFailure(Call<customer_location> call, Throwable t) {
                    Log.e(TAG, "Unable to submit post to API.");
                }
            });}
    }

    private boolean isOnline() {
        return networkStateChangeReceiver.isNetworkAvailable(mContext);
    }
}

File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperClienteGPS.kt
package com.friendlysystemgroup.friendlypos.principal.helpers

import android.content.ContentValues
import android.content.Context
import android.util.Log
import android.widget.Toast
import com.friendlysystemgroup.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver
import com.friendlysystemgroup.friendlypos.application.datamanager.BaseManager.Companion.api
import com.friendlysystemgroup.friendlypos.application.interfaces.RequestInterface
import com.friendlysystemgroup.friendlypos.login.util.SessionPrefes.Companion.get
import com.friendlysystemgroup.friendlypos.principal.activity.MenuPrincipal
import com.friendlysystemgroup.friendlypos.principal.modelo.EnviarClienteGPS
import com.friendlysystemgroup.friendlypos.principal.modelo.customer_location

/**
 * Created by DelvoM on 24/10/2018.
 */
class SubirHelperClienteGPS(private val activity: MenuPrincipal) {
    private val networkStateChangeReceiver =
        NetworkStateChangeReceiver()

    private val mContext: Context = activity
    private val mAPIService: RequestInterface? = api

    var codigo: Int = 0
    var respuestaServer: String? = null
    var codigoS: String? = null
    var mensajeS: String? = null
    var resultS: String? = null

    fun sendPostClienteGPS(facturaQuery: EnviarClienteGPS) {
        val token = "Bearer ${get(mContext).token}"
        Log.d("tokenCliente", token)
        
        if (isOnline) {
            Log.d("factura1", "$facturaQuery")
            
            mAPIService?.savePostClienteGPS(facturaQuery, token)
                ?.enqueue(object : retrofit2.Callback<customer_location?> {
                    override fun onResponse(
                        call: retrofit2.Call<customer_location?>?,
                        response: retrofit2.Response<customer_location?>
                    ) {
                        if (response.isSuccessful) {
                            response.body()?.let { body ->
                                Log.d("respVentaDirecta", body.toString())
                                
                                codigo = response.code()
                                codigoS = body.code
                                mensajeS = body.message
                                resultS = body.isResult.toString()

                                activity.codigoDeRespuestaClienteGPS(
                                    codigoS ?: "",
                                    mensajeS,
                                    resultS ?: "",
                                    codigo
                                )
                            }
                        } else {
                            Toast.makeText(mContext, "ERRRRROOOORRRR", Toast.LENGTH_SHORT).show()
                        }
                    }

                    override fun onFailure(
                        call: retrofit2.Call<customer_location?>?,
                        t: Throwable?
                    ) {
                        Log.e(ContentValues.TAG, "Unable to submit post to API.")
                    }
                })
        }
    }

    private val isOnline: Boolean
        get() = networkStateChangeReceiver.isNetworkAvailable(mContext)
}
